### ISSUE TITLE
Media library can only be opened once - upload freezes on second opening

### ISSUE THREAD

**Issue by rowroll** [WordPress/gutenberg#11373]
I keep having to refresh the whole editor page when I open the media library upload for the second time.  I can see the upload for a new photo start, but it then stops, and just hangs.  Closing the media library and opening it does not help, I have to refresh the whole page.

**Comment by designsimply** [comment]
I was unable to reproduce the problem using the following testing steps and testing with WordPress 4.9.8 and Gutenberg 4.2.0 using Firefox 63.0.1 on macOS 10.13.6:

1. Go to Posts > Add new.
1. Add a title.
1. Insert an image using the "Media Library" button in the image block.
1. Insert another image using the "Media Library" button in the image block.
1. Add some text in a paragraph.
1. Insert another image using the "Media Library" button in the image block.

<strong>Result:</strong> the media library opened normally for me when using the "Media Library" button in the image block. ([58s](https://cloudup.com/ciVfjC8O4zI))

Are your steps the same as the ones I've noted above?

Are you using a classic block (or some other block) instead of an image block?

Are you able to open the Media Library directly using that option in the left sidebar?

Can you note your OS and browser versions for reference?

If possible, can you try to see if there are any errors in the browser console at the time you try to open the media library for the second time?

**Comment by rowroll** [comment]
Hi,
I have a Windows 10 machine, running Chrome Version 70.0.3538.102 (Official Build) (64-bit).  I can't seem to find the version number of Gutenberg, but I first installed it the day I submitted the comment.

Here are my steps:
1. Go to Posts > Add new.
2. Add a title.
3. Add a Gallery block.  
4.  Click on Media Library inside the Gallery Block, add two photos.  
5. Enter the gallery on to the page.  The photos appear.  All is fine.
6. Add another Gallery block.
7. Click on Media Library in the Gallery Block again, 
8.  Drag a photo onto the Media library to upload it. 
9.  The upload bar shows 15%-ish done, but then just hangs.  Nothing is shown on the far right side.  If I click on the photo upload, I get the name of the image, so the Media library is only partially not working.
10.  Using the x on the upper right to close the Media library and opening it again, just brings up the same frozen upload.
11.  Drag another photo into the Media Library, upload also hangs.
12. To resolve I have to refresh the entire page.  Then I can resume uploading via the Media library.

The developer console has the following message
Active resource loading counts reached to a per-frame limit while the tab is in background. Network requests will be delayed until a previous loading finishes, or the tab is foregrounded. See <URL> for more details
load-scripts.php?c=0&load[]=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=4.9.8:9 JQMIGRATE: Migrate is installed, version 1.4.1
media-views.min.js?ver=4.9.8:2 Uncaught TypeError: Cannot read property 'text' of undefined
    at n.info (media-views.min.js?ver=4.9.8:2)
    at _ (backbone.min.js?ver=1.2.3:1)
    at m (backbone.min.js?ver=1.2.3:1)
    at f (backbone.min.js?ver=1.2.3:1)
    at n.u.trigger (backbone.min.js?ver=1.2.3:1)
    at n._onModelEvent (backbone.min.js?ver=1.2.3:1)
    at _ (backbone.min.js?ver=1.2.3:1)
    at m (backbone.min.js?ver=1.2.3:1)
    at f (backbone.min.js?ver=1.2.3:1)
    at n.u.trigger (backbone.min.js?ver=1.2.3:1)

Let me know if you need more info.

**Comment by bobbingwide** [comment]
I experienced the same problem with a similar scenario; attempting to upload and set the featured image. Having failed to make any progress while attempting to debug my scenario I attempted to reproduce the problem with the steps listed above.  The console log , using WordPress 5.0.2 and Gutenberg v4.7.1, is almost identical.

```
Uncaught TypeError: Cannot read property 'text' of undefined
    at n.info (media-views.min.js?ver=5.0.2:2)
    at _ (backbone.min.js?ver=1.2.3:1)
    at m (backbone.min.js?ver=1.2.3:1)
    at f (backbone.min.js?ver=1.2.3:1)
    at n.u.trigger (backbone.min.js?ver=1.2.3:1)
    at n._onModelEvent (backbone.min.js?ver=1.2.3:1)
    at _ (backbone.min.js?ver=1.2.3:1)
    at m (backbone.min.js?ver=1.2.3:1)
    at f (backbone.min.js?ver=1.2.3:1)
    at n.u.trigger (backbone.min.js?ver=1.2.3:1)
```
![image](https://user-images.githubusercontent.com/2474435/50658522-91111080-0f91-11e9-9271-1143b14982da.png)

**Comment by bobbingwide** [comment]
I believe that I was unable to reproduce the problem with WordPress SEO deactivated.

**Comment by bobbingwide** [comment]
I was also able to reproduce the problem with define('SCRIPT_DEBUG', true ); in WordPress 5.0.2 with no plugins and the Twenty Nineteen theme v1.1
```
Uncaught TypeError: Cannot read property 'text' of undefined
    at n.info (media-views.js?ver=5.0.2:5015)
    at _ (backbone.min.js?ver=1.2.3:1)
    at m (backbone.min.js?ver=1.2.3:1)
    at f (backbone.min.js?ver=1.2.3:1)
    at n.u.trigger (backbone.min.js?ver=1.2.3:1)
    at n._onModelEvent (backbone.min.js?ver=1.2.3:1)
    at _ (backbone.min.js?ver=1.2.3:1)
    at m (backbone.min.js?ver=1.2.3:1)
    at f (backbone.min.js?ver=1.2.3:1)
    at n.u.trigger (backbone.min.js?ver=1.2.3:1)
```
I added `console.log( this );` to the info: function. This screen capture might help.
![image](https://user-images.githubusercontent.com/2474435/50662029-ed2d6200-0f9c-11e9-9447-0200483c4f60.png)

**Comment by thmplt** [comment]
having the exact same issue. I have posts that are 100% relying on the galleries. Has anyone come up with a patch? or any info when WP will fix this?

**Comment by rowroll** [comment]
This is still an issue, just confirmed it in Gutenberg 5.4.0 on Windows 10 and Chrome Version 73.0.3683.103 (Official Build) (64-bit).

**Comment by flexerd** [comment]
If this helps in advancing this in Cue having the same exact issue.

**Comment by rafaucau** [comment]
Still nothing going on in this case?

**Comment by jakewhiteley** [comment]
I got this, but on an ACF settings page. I'm probably being stupid, but I cannot find a ticket for it on Trac. Does anyone know if its even being looked at by the core team?

**Comment by darwin370** [comment]
Same problem here with WordPress 5.6.1, Twenty Twenty-One 1.1, no plugins, Chrome Version 88.0.4324.150 (Official Build) (64-Bit), reproducable with steps provided by @rowroll

> 1. Go to Posts > Add new.
> 2. Add a title.
> 3. Add a Gallery block.
> 4. Click on Media Library inside the Gallery Block, add two photos.
> 5. Enter the gallery on to the page.  The photos appear.  All is fine.
> 6. Add another Gallery block.
> 7. Click on Media Library in the Gallery Block again,
> 8. Drag a photo onto the Media library to upload it.
> 9. The upload bar shows 15%-ish done, but then just hangs.  ...

`Uncaught TypeError: Cannot read property 'text' of undefined
    at n.info (media-views.min.js?ver=5.6.1:2)
    at p (backbone.min.js?ver=1.4.0:2)
    at f (backbone.min.js?ver=1.4.0:2)
    at c (backbone.min.js?ver=1.4.0:2)
    at n.trigger (backbone.min.js?ver=1.4.0:2)
    at n._onModelEvent (backbone.min.js?ver=1.4.0:2)
    at p (backbone.min.js?ver=1.4.0:2)
    at f (backbone.min.js?ver=1.4.0:2)
    at c (backbone.min.js?ver=1.4.0:2)
    at n.trigger (backbone.min.js?ver=1.4.0:2)
`

**Comment by paaljoachim** [comment]
I tested using WordPress 5.7.
Gutenberg plugin 10.
Twenty Twenty One.
Chrome

> 1. Go to Posts > Add new.
> 2. Add a title.
> 3. Add a Gallery block.
> 4. Click on Media Library inside the Gallery Block, add two photos.
> 5. Enter the gallery on to the page. The photos appear. All is fine.
> 6. Add another Gallery block.
> 7. Click on Media Library in the Gallery Block again,
> 8. Drag a photo onto the Media library to upload it.
> 9. The upload bar shows 15%-ish done, but then just hangs. Nothing is shown on the far right side. If I click on the photo upload, I get the name of the image, so the Media library is only partially not working.
> 10. Using the x on the upper right to close the Media library and opening it again, just brings up the same frozen upload.
> 11. Drag another photo into the Media Library, upload also hangs.
> 12. To resolve I have to refresh the entire page. Then I can resume uploading via the Media library.

I tested from 1 up to 9. Uploading gets stuck as you mention. This is what I noticed.

![Screen Shot 2021-02-21 at 10 51 12](https://user-images.githubusercontent.com/5323259/108621467-0b7fa180-7433-11eb-9915-2e0c6f6fc911.png)

I am also getting this error along the way: https://github.com/WordPress/gutenberg/issues/25805

The Block Gallery is getting renewed in this PR: https://github.com/WordPress/gutenberg/pull/25940

@glendaviesnz 
Seems like we need to check this out in relation to the new PR.

**Comment by rebeccahum** [comment]
I'm running into this issue with WP 5.7 currently.  It is reproducible on a local install with the above test steps.

**Comment by glendaviesnz** [comment]
> @glendaviesnz
> Seems like we need to check this out in relation to the new PR.

Thanks for the ping - looks like this is an issue in the core wp media library which the gallery refactor doesn't touch, so the gallery refactor will have the same issue, and any fix in core wp for this will fix the issue in the gallery refactor.

**Comment by lyx** [comment]
I hope this info can narrow down the issue - I did some testing as well.

It seems that this upload freeze only occurs after a gallery component has had images inserted into the component, by clicking the `Insert Gallery` button. 

Theoretically, based on the test cases here and my testing, it stands to reason that something is affecting the uploading of images when images are inserted into a gallery. I'll update here with any findings.

**Comment by glendaviesnz** [comment]
I have added a core ticket for this https://core.trac.wordpress.org/ticket/53169 and possible fix https://github.com/WordPress/wordpress-develop/pull/1235

**Comment by glendaviesnz** [comment]
This is resolved in WP 5.8 and gutenberg 11.3.
